src/api/c++/z3++.h

```diff
@@ -85,6 +86,10 @@ namespace z3 {
         std::string m_msg;
     public:
         exception(char const * msg):m_msg(msg) {}
```

Comment:
Rule of 5; the destructor was declared

Reply:
these definitions are useless

Reply:
They allow different methods of catching the exception. Since the destructor was declared, that prevents the move operator and constructor from being implicitly generated, thus we default them. But since we do, that prevents the compiler from generating the copy operator / constructor. This is known as the rule of 5. Since the message is a string and member variable, it will benefit from having move construction / assignment.

Reply:
the destructor can be removed; it's already declared in std::exception

Reply:
Ok, I will apply the rule of 0 then.